src/adapter/src/coord.rs

```diff
@@ -1056,6 +1057,23 @@ impl Coordinator {
         // state with what it has learned from the adapter.
         self.controller.storage.reconcile_state().await;
 
+        // Cleanup orphaned secrets. Errors during list() or delete() do not
+        // need to prevent bootstrap from succeeding; we will retry next
+        // startup.
+        if let Ok(controller_secrets) = self.secrets_controller.list().await {
+            let user_secrets: BTreeSet<GlobalId> =
+                self.catalog.secrets().map(|entry| entry.id()).collect();
```

Comment:
So this is weird, but secrets can be either:

1. Associated with a secret object in the catalog.
2. Associated with a connection object (specifically an SSH tunnel connection, but maybe other connection types in the future).

So I think you want to get IDs of secrets _and_ connections here, and that'll fix the SSH test.

Reply:
Indeed. Removing `id.is_user()` from the filter didn't help. Maybe `.is_secret()` should be taught about connections? This API is real easy to misuse.

Reply:
No, `is_secret` should unambiguously refer to whether the entry is a secret in the catalog. The conflation occurs at a lower layer (the secrets controller layer), in which a secrets controller secret is associated with a catalog connection.

Reply:
Made a `fn secrets_and_connections` that passes tests. Should be ok now.

Reply:
Ah much better. Done.